walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,83 @@
 
 package com.google.sps;
 
-import java.util.Collection;
+import java.util.*;
 
 public final class FindMeetingQuery {
-  public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
-  }
+
+    /**
+    * Returns a set of times in a day when a specified meeting could happen, given 
+    * known events during that day.
+    */
+    public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
+        Collection<TimeRange> includeOptional = query(events, request, true);
```

Comment:
What does "optional" mean here?

Reply:
It means event slots are found including optional attendees. I'll change the name since it's the same as the bool I use lower down. Done in ffce723cd971d8b9d4b20498d1e2a8273a380a00.